src/robot.cpp

```diff
@@ -52,18 +52,29 @@ Robot::Robot(const std::string &name, Id id, bool disable_speedcap, bool uncerta
     laser_pose.t.z = 0.3;
 
     nh = ros::NodeHandle();
+    //parameters
+    std::string laser_param, odom_param, bumper_f_param, bumper_b_param, base_ref_param, open_door_param, speak_param, play_param;
+    if (!nh.getParam("laser_", laser_param)) {ROS_ERROR_STREAM("Parameter " << "laser_" << " not set"); exit(EXIT_FAILURE);};
+    if (!nh.getParam("odom_", odom_param)) {ROS_ERROR_STREAM("Parameter " << "odom_" << " not set"); exit(EXIT_FAILURE);};
+    if (!nh.getParam("bumper_f_", bumper_f_param)) {ROS_ERROR_STREAM("Parameter " << "bumper_f_" << " not set"); exit(EXIT_FAILURE);};
+    if (!nh.getParam("bumper_b_", bumper_b_param)) {ROS_ERROR_STREAM("Parameter " << "bumper_b_" << " not set"); exit(EXIT_FAILURE);};
+    //if (!nh.getParam("base_ref_", base_ref_param)) {ROS_ERROR_STREAM("Parameter " << "base_ref_" << " not set");};
+    if (!nh.getParam("open_door_", open_door_param)) {ROS_ERROR_STREAM("Parameter " << "open_door_" << " not set");};
+    if (!nh.getParam("speak_", speak_param)) {ROS_ERROR_STREAM("Parameter " << "speak_" << " not set");};
```

Comment:
This means we always have to set all parameters, correct? 
Why don't we use default values for convenience, a lot of them have straightforward names such as /odom

Reply:
Because that would cause invisible errors. I. E. Robot object initialising but not doing anything